src/service/persister.ts

```diff
@@ -162,13 +168,59 @@ export class Persister<T extends Persistable> implements ILoadAll<T> {
     public persist = (report: T) => {
         this.collection.then(coll => {
             this._saver(report);
+            if (["fv","md","msg","mt","pos","qp-sub","tbp","osr","tsv"].indexOf(this._dbName)>-1)
+              coll.deleteMany({ time: { $exists:true } }, err => {
+                  if (err)
+                      this._log.error(err, "Unable to deleteMany", this._dbName, report);
+              });
             coll.insertOne(report, err => {
                 if (err)
                     this._log.error(err, "Unable to insert", this._dbName, report);
             });
         }).done();
     };
 
+    public perfind = (report: T, side: Models.Side, width?: number, price?: number): any => {
+        var deferred = Q.defer<T[]>();
+        this.collection.then(coll => {
+            coll.find({ $and: [
+              { price: side==Models.Side.Bid?{ $gt: width+price }:{ $lt: price-width } },
+              { side: side==Models.Side.Bid?1:0 },
+              { $where: "this.quantity - this.alloc > 0" }
+            ] }).limit(10000).project({ _id: 0 }).sort({ alloc: 1, price: side==Models.Side.Bid?1:-1 })
+            .toArray((err, arr) => {
+                if (err) {
+                    deferred.reject(err);
+                }
+                else if (arr.length === 0) {
+                    deferred.resolve(null);
+                }
+                else {
+                    _.forEach(arr, this._loader);
+                    deferred.resolve(arr);
+                }
+            });;
+        }).done();
+
+        return deferred.promise;
+    };
+
+    public repersist = (report: T, trade: Models.Trade) => {
```

Comment:
I am very against these changes to persister.ts. The data, once persisted, should never change and data in the database should always be immutable. This is good from a code-quality standpoint (reduces cohesion between persister and the quoting logic) but also from a data analysis standpoint (I can recreate the state of the world at any time solely from whats in the DB while performing my analysis)

If there needs to be calculations done to the persisted data, the code should fetch it from the DB, calculate the new value and use it if needed, then persist it back out into a new table/collection if it's even needed.

The interface of IPersist<T>/ILoad*<T> should not need to know about the implementations of various quoting styles or really even what data is being flung at them. The addition of trade and side to the interface violates that.

In this case, I would recommend the approach of having a second collection of closed trade IDs, having a secondary component figure out which trades are opened and closed, then persist out what's actually closed.